compiler/src/dotty/tools/dotc/fromtasty/ReadTasty.scala

```diff
@@ -44,9 +44,8 @@ class ReadTasty extends Phase {
                 Some(unit)
               }
             case tree: Tree[?] =>
+              // TODO handle correctly this case correctly to get the tree or avoid it completely.
               cls.denot.infoOrCompleter match {
-                case _: NoLoader => Some(Scala2CompilationUnit(cls.fullName.toString))
-                case _ if cls.flags.is(Flags.JavaDefined) => Some(JavaCompilationUnit(cls.fullName.toString))
                 case _ => Some(AlreadyLoadedCompilationUnit(cls.denot.fullName.toString))
               }
```

Comment:
The `match` is kept intentionally? Semantically, it can be removed.

Reply:
We should be able to remove it. But last time I tried there was a failure with some of the `-from-tasty` tests. As this is unrelated to the public API I left it for a second PR.

Reply:
I also left the comment above to not forget to remove this